src/common/meta/src/key.rs

```diff
@@ -403,30 +400,28 @@ impl TableMetadataManager {
             .table_info_manager()
             .build_create_txn(table_id, &table_info_value)?;
 
-        // Creates datanode table key value pairs.
-        let distribution = region_distribution(&region_routes)?;
-        let create_datanode_table_txn = self.datanode_table_manager().build_create_txn(
-            table_id,
-            &engine,
-            &region_storage_path,
-            region_options,
-            region_wal_options,
-            distribution,
-        )?;
-
-        // Creates table route.
-        let table_route_value = TableRouteValue::new(region_routes);
         let (create_table_route_txn, on_create_table_route_failure) = self
             .table_route_manager()
             .build_create_txn(table_id, &table_route_value)?;
 
-        let txn = Txn::merge_all(vec![
+        let mut txn = Txn::merge_all(vec![
             create_table_name_txn,
             create_table_info_txn,
-            create_datanode_table_txn,
             create_table_route_txn,
         ]);
 
+        if let TableRouteValue::Physical(x) = &table_route_value {
+            let create_datanode_table_txn = self.datanode_table_manager().build_create_txn(
+                table_id,
+                &engine,
+                &region_storage_path,
+                region_options,
+                region_wal_options,
+                region_distribution(&x.region_routes)?,
+            )?;
+            txn = txn.merge(create_datanode_table_txn);
+        }
```

Comment:
Should we register the logical table route for the metric table? @waynexia Maybe we need to add a TODO here.

Reply:
Do logical regions need to open explicitly on datanodes? I think it's better not to, make them opened already by opening the physical regions. This way we can truly untie the relationship between datanode and logical region.

Reply:
Oh, my mistake. I mixed the `DatanodeTableValue` and `TableRouteValue`.